sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/ViewportStatus.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.maps.plugin.viewport.experimental
+
+import com.mapbox.maps.plugin.viewport.experimental.state.ViewportState
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransition
+
+/**
+ * Represents the status of the viewport.
+ *
+ * It could be either a [ViewportState] or [ViewportTransition].
+ */
+sealed class ViewportStatus {
+  /**
+   * Represents the current status is a [ViewportState].
+   *
+   * The state is null if current status is IDLE.
+   */
+  class State(val state: ViewportState?) : ViewportStatus() {
```

Comment:
It'd be great to add equals and other overrides to this class.

Reply:
+1. On iOS, I implemented equals and hash based on the identity of the member objects.

Reply:
Added overrides for equals and hash.